runtime-modules/bureaucracy/src/lib.rs

```diff
@@ -835,6 +856,93 @@ decl_module! {
             // Trigger event
             Self::deposit_event(RawEvent::WorkerOpeningFilled(worker_opening_id, worker_application_id_to_worker_id));
         }
+
+        // ****************** Worker stakes **********************
+
+        /// Slashes the worker stake, demands a leader origin. No limits, no actions on zero stake.
+        /// If slashing balance greater than the existing stake - stake is slashed to zero.
+        pub fn slash_worker_stake(origin, worker_id: WorkerId<T>, balance: BalanceOf<T>) {
+            Self::ensure_origin_is_set_lead(origin)?;
+
+            let worker = Self::ensure_worker_exists(&worker_id)?;
+
+            ensure!(balance != <BalanceOf<T>>::zero(), Error::StakeBalanceCannotBeZero);
+
+            let stake_profile = worker.role_stake_profile.ok_or(Error::NoWorkerStakeProfile)?;
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            ensure_on_wrapped_error!(
+                <stake::Module<T>>::slash_immediate(
+                    &stake_profile.stake_id,
+                    balance,
+                    false
+                )
+            )?;
+
+            Self::deposit_event(RawEvent::WorkerStakeSlashed(worker_id));
+        }
+
+        /// Decreases the worker stake and returns the remainder to the worker role_account,
+        /// demands a leader origin. Can be decreased to zero, no actions on zero stake.
+        pub fn decrease_worker_stake(origin, worker_id: WorkerId<T>, balance: BalanceOf<T>) {
+            Self::ensure_origin_is_set_lead(origin)?;
+
+            let worker = Self::ensure_worker_exists(&worker_id)?;
+
+            ensure!(balance != <BalanceOf<T>>::zero(), Error::StakeBalanceCannotBeZero);
+
+            let stake_profile = worker.role_stake_profile.ok_or(Error::NoWorkerStakeProfile)?;
+
+            ensure_on_wrapped_error!(
+                <stake::Module<T>>::ensure_can_decrease_stake(&stake_profile.stake_id, balance)
```

Comment:
Is this call actually required, what does it do that is not already covered in `decrease_stake_to_account`?

Reply:
Actually, decrease_stake_to_account() should call the ensure_can_decrease_stake() internally. But it doesn't. But even if it would do that we use 'Parity recommended' style to use 'ensure_*' methods before mutation methods.

Reply:
1) Are you saying there is currently a bug in `decrease_stake` then? is it not making all required checks itself?

2) I don't think it makes sense to do if it does not add anything, just makes it harder to see what is going on. When we make calls to other methods in our own library stack we often end up doing this.